File: app/backend/app/schemas/hot_board.py
"""首页今日热榜响应契约。"""

from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field


HotBoardPanelId = Literal["hot_discuss", "finance_lookup", "market_view"]


class HotBoardPanel(BaseModel):
    id: HotBoardPanelId
    title: str
    items: list[str] = Field(default_factory=list, min_length=1, max_length=5)


class HotBoardResponse(BaseModel):
    as_of: str
    source: Literal["web", "fallback"]
    panels: list[HotBoardPanel]


__all__ = ["HotBoardPanel", "HotBoardPanelId", "HotBoardResponse"]

File: app/frontend/src/services/api/hotBoard.ts
import { apiFetch } from './client'

export type HotBoardPanelId = 'hot_discuss' | 'finance_lookup' | 'market_view'

export interface HotBoardPanel {
  id: HotBoardPanelId
  title: string
  items: string[]
}

export interface HotBoardResponse {
  as_of: string
  source: 'web' | 'fallback' | 'stale'
  panels: HotBoardPanel[]
}

export function fetchHotBoard(refresh = false): Promise<HotBoardResponse> {
  const query = refresh ? '?refresh=true' : ''
  return apiFetch<HotBoardResponse>(`/api/hot-board${query}`)
}
